src/couch_replicator/src/couch_replicator_api_wrap.erl

```diff
@@ -68,20 +68,24 @@ db_uri(Db) ->
 
 
 db_open(Db, Options) ->
-    db_open(Db, Options, false).
+    db_open(Db, Options, false, []).
 
-db_open(#httpdb{} = Db1, _Options, Create) ->
+db_open(#httpdb{} = Db1, _Options, Create, CreateParams) ->
     {ok, Db} = couch_replicator_httpc:setup(Db1),
     try
         case Create of
         false ->
             ok;
         true ->
-            send_req(Db, [{method, put}],
+            Db2 = case couch_util:get_value(<<"q">>, CreateParams) of
```

Comment:
We explicitly handle `q` parameter here, I wonder if it is worth making it more generic. What if users want to specify `n`, `placement` or even the new PSE `engine` parameters? I think it should be possible. Or do you think it's better to restrict it to specific params only?

Reply:
Yes, I extend this to be generic.